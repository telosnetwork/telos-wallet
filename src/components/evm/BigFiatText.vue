<script lang="ts">
import { defineComponent } from 'vue';
import { prettyPrintCurrency } from 'src/antelope/stores/utils';
import { useUserStore } from 'src/antelope';

export default defineComponent({
    name: 'BigFiatText',
    props: {
        amount: {
            type: Number,
            required: true,
        },
    },
    computed: {
        prettyAmount() {
            const { fiatLocale, fiatCurrency } = useUserStore();

            return prettyPrintCurrency(
                this.amount,
                2,
                fiatLocale,
                false,
                fiatCurrency,
            );
        },
    },
});
</script>

<template>
<div class="c-big-fiat-text">
    {{ prettyAmount }}
</div>
</template>

<style lang="scss">
.c-big-fiat-text {
    line-height: 48px;
    font-weight: 600;
    font-size: 32px;
    text-align: center;

    @media screen and (min-width: 360px) {
        font-size: 40px;
    }
}
</style>
