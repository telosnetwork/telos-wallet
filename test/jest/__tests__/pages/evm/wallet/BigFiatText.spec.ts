import { shallowMount } from '@vue/test-utils';

jest.mock('src/antelope', () => ({
    useUserStore: () => ({
        fiatLocale: 'en-US',
        fiatCurrency: 'USD',
    }),
}));

import BigFiatText from 'components/evm/BigFiatText.vue';
describe('BigFiatText.vue', () => {
    it('should have the correct name', () => {
        expect(BigFiatText.name).toBe('BigFiatText');
    });

    it('should render correctly', () => {
        const wrapper = shallowMount(BigFiatText, {
            props: {
                amount: 12345,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
