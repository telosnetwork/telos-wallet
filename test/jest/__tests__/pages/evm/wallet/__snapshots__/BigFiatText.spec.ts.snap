// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigFiatText.vue should render correctly 1`] = `
<div
  class="c-big-fiat-text"
>
  $12,345.00
</div>
`;
